blas/unit_test/Test_Blas2_ger.hpp

```diff
@@ -0,0 +1,1616 @@
+         , ScalarY
+         , tLayoutY
+         , ScalarA
+         , tLayoutA
+         , Device
+         >::~GerTester()
+{
+  // Nothing to do
+}
+
+template <class ScalarX, class tLayoutX, class ScalarY, class tLayoutY, class ScalarA, class tLayoutA, class Device>
+void GerTester< ScalarX
+              , tLayoutX
+              , ScalarY
+              , tLayoutY
+              , ScalarA
+              , tLayoutA
+              , Device
+              >::test( const int  M
+                     , const int  N
+                     , const int  nonConstConstCombinations
+                     , const bool useAnalyticalResults
+                     , const bool useHermitianOption
+                     )
+{
+  std::cout << "Entering GerTester::test()... - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - " << std::endl;
+
+  std::cout << "_A_is_complex = "                     << _A_is_complex
+            << ", _A_is_lr = "                        << _A_is_lr
+            << ", _A_is_ll = "                        << _A_is_ll
+            << ", _testIsGpu = "                      << _testIsGpu
+            << ", _vanillaUsesDifferentOrderOfOps = " << _vanillaUsesDifferentOrderOfOps
+            << ", _epsAbs = "                         << _epsAbs
+            << ", _epsRel = "                         << _epsRel
+            << std::endl;
+  
+  // ********************************************************************
+  // Step 1 of 9: declare main types and variables
+  // ********************************************************************
+  _M = M;
+  _N = N;
+  _useAnalyticalResults = useAnalyticalResults;
+  _useHermitianOption   = useHermitianOption;
+
+#ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+  _kkGerShouldThrowException = false;
+  if (_A_is_complex && _useHermitianOption) {
+    if ((_testIsGpu == false) &&
+        (_A_is_ll   == false)) {
+      _kkGerShouldThrowException = true;
+    }
+    else if ((_testIsGpu == true ) &&
+             (_A_is_ll   == false)) {
+      _kkGerShouldThrowException = true;
+    }
+  }
+#endif
+
+  bool test_x_y  (false);
+  bool test_cx_y (false);
+  bool test_x_cy (false);
+  bool test_cx_cy(false);
+  if (nonConstConstCombinations == 0) {
+    test_x_y = true;
+  }
+  else if (nonConstConstCombinations == 1) {
+    test_cx_y = true;
+  }
+  else if (nonConstConstCombinations == 2) {
+    test_x_cy = true;
+  }
+  else if (nonConstConstCombinations == 3) {
+    test_cx_cy = true;
+  }
+  else {
+    test_x_y   = true;
+    test_cx_y  = true;
+    test_x_cy  = true;
+    test_cx_cy = true;
+  }
+
+  _ViewTypeX x("X", _M);
+  _ViewTypeY y("Y", _N);
+  _ViewTypeA A("A", _M, _N);
+
+  typename _ViewTypeX::const_type c_x = x;
+  typename _ViewTypeY::const_type c_y = y;
+
+  _HostViewTypeX h_x = Kokkos::create_mirror_view(x);
+  _HostViewTypeY h_y = Kokkos::create_mirror_view(y);
+  _HostViewTypeA h_A = Kokkos::create_mirror_view(A);
```

Comment:
I think this might lead to false negative on Host since h_A will be modified in Steps 2 and 3?

Reply:
Done. Thanks for the hint.